Sources/TurboNavigationDelegate.swift

```diff
@@ -10,9 +10,14 @@ public protocol TurboNavigationDelegate: AnyObject {
     /// Respond to authentication challenge presented by web servers behing basic auth.
     func didReceiveAuthenticationChallenge(_ challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void)
 
-    /// Optional. Implement to override or customize the controller to be displayed.
-    /// Return `nil` to not display or route anything.
-    func controller(_ controller: VisitableViewController, forProposal proposal: VisitProposal) -> UIViewController?
+    /// Optional. Allow or cancel a visit.
+    /// If allowed, you may provide a view controller to be displayed, otherwise a new `VisitableViewController` is used.
+    /// If rejected, no changes to navigation occur.
+    /// If not implemented, proposals are accepted and a new `VisitableViewController` is displayed.
+    ///
+    /// - Parameter proposal: navigation destination
+    /// - Returns: how to react to the visit proposal
+    func response(forProposal proposal: VisitProposal) -> VisitProposalResponse
```

Comment:
I think we should come up with a different name for this. A `response` is language we use for an actual visit/http responses, which would be confusing long-term for the result of a proposal.

Reply:
I'd like to convey that this isn't about what to display, but about whether to display anything at all given a VisitProposal. What do you think of...

`shouldVisit(_ proposal: VisitProposal) -> VisitDecision`?

Reply:
That's definitely an improvement. Would it make sense to split up the decision whether you should navigate and obtaining the controller instance?

For example, we could have:
```
shouldVisit(_ proposal: VisitProposal): Bool // or a simple enum if it makes sense
```

Then, if the app determines you should continue the visit, call something like:
```
controller(for controllerName: String): UIViewController
```

The `controllerName` would be obtained from the path config `controller-name` value. (Sorry, I know I'm jumping ahead to a future PR that we discussed, but it seemed relevant here.)

Reply:
Yes. I wanted to keep it in one function, but I think I'm just forcing the issue a bit too much. 

I'll add `shouldVisit(_ proposal: VisitProposal): Bool`, and then change`controller(_, forProposal:)`'s return value from `UIViewController?` to `UIViewController`. Then I can build the `controller-name` on top of this.

Reply:
Cool, that sounds like a great idea to me 👍

Reply:
I prefer to keep everything in a single function. It should make it easer to get started with Turbo Native, even if the naming might not be perfect.

What about something like this?

```swift
enum ProposalResult {
    case accept
    case custom(UIViewController)
    case reject
}

func handle(proposal: VisitProposal) -> ProposalResult
```

Reply:
This is a nice improvement and easy to understand 👍 The only downside I see is we then can't codify the `view-controller` path config lookup convention through the API explicitly, but maybe that's ok.

Reply:
I've used Joe's suggestion, but I've slightly modified case names:

````swift
enum ProposalResult {
    case accept
    case acceptCustom(UIViewController)
    case reject
}
````

I will take a look at the `view-controller` option in a follow-up PR.